examples/test_method/Form_ExitStruct.rb

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env ruby
+
+require 'rubygems'
+require "newt"
+
+begin
+
+  Newt::Screen.new
+
+  b1 = Newt::Button.new(1, 5, "Button1")
+  b2 = Newt::Button.new(1, 9, "Button2")
+
+  b = Newt::Button.new(1, 13, "Exit")
+
+  f = Newt::Form.new
+  f.add_hotkey(?\e.ord)
+  f.add(b1, b2, b)
+
+  rv = f.run()
+
+ensure
+  Newt::Screen.finish
+end
+
+puts rv.inspect
+puts "Button1" if rv == b1
+puts "Button2" if b2 == rv
+puts "Exit" if rv == b
+puts "Escape" if (rv.reason == Newt::EXIT_HOTKEY && rv.key == ?\e.ord)
```

Comment:
I suppose that could work. I don't know if I could contribute any unit testing at this time. I considered adding a new method to Form that calls newtFormRun(). I could do that if you prefer to maintain compatibility that way. I'm not sure what to call it though.

I'm mainly just adding these features for internal use at my job and figured I'd contribute them back.